hasher-matcher-actioner/hmalib/lambdas/api/exchanges.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+APIs to power viewing and editing of exchanges.
+"""
+
+import bottle
+
+from threatexchange.exchanges import auth
+
+from hmalib.lambdas.api.middleware import SubApp
+from hmalib.aws_secrets import AWSSecrets
+from hmalib.common.config import HMAConfig
+from hmalib.common.configs.tx_apis import (
+    ToggleableSignalExchangeAPIConfig,
+    add_signal_exchange_api,
+    set_status_signal_exchange_api,
+)
+
+
+def get_exchanges_api(hma_config_table: str, secrets_prefix: str) -> bottle.Bottle:
+    exchanges_api = SubApp()
+    HMAConfig.initialize(hma_config_table)
+    secrets = AWSSecrets(secrets_prefix)
+
+    @exchanges_api.get("/")
+    def get_exchanges():
+        """
+        Get the currently enabled APIs.
+        """
+        apis = ToggleableSignalExchangeAPIConfig.get_all()
+        return {
+            "exchanges": {
+                api.signal_exchange_api_class: {
+                    "enabled": api.enabled,
+                    "supports_credentials": issubclass(
+                        api.to_concrete_class(), auth.SignalExchangeWithAuth
+                    ),
+                }
+                for api in apis
+            }
+        }
+
+    @exchanges_api.post("/update-enabled")
+    def update_enabled():
+        classname = bottle.request.query.get("class")
+        target_status = bottle.request.query.get("enabled") == "true"
+
+        return {"result": set_status_signal_exchange_api(classname, target_status).name}
+
+    @exchanges_api.post("/add-new-exchange")
+    def add_new_exchange():
+        classname = bottle.request.query.get("class")
+        return {"result": add_signal_exchange_api(classname).name}
+
+    @exchanges_api.get("/get-credential-string")
+    def get_credential_string():
+        classname = bottle.request.query.get("class")
+        api = [
+            x
+            for x in ToggleableSignalExchangeAPIConfig.get_all()
+            if x.signal_exchange_api_class == classname
+        ]
+
+        if not api:
+            return {"result": "failed"}
+
+        try:
+            return {
+                "credential_string": secrets.get_secret(api[0].get_credential_name())
+            }
+        except ValueError:
+            return {"result": "failed"}
+
+    @exchanges_api.post("/set-credential-string")
+    def set_credential_string():
+        classname = bottle.request.json.get("class")
+        credential_string = bottle.request.json.get("credential_string")
+
+        api = [
+            x
+            for x in ToggleableSignalExchangeAPIConfig.get_all()
+            if x.signal_exchange_api_class == classname
+        ]
```

Comment:
(ignorable): There a compact comprehension with `next((<the same comprehension>), None)` you can use, but this is fine as well.

The way this reads makes me ask about len() > 1 since you aren't handling it speicifically